src/Service/ServiceEndpoint.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Basis\Nats\Service;
+
+use Basis\Nats\Client;
+use Basis\Nats\Message\Payload;
+use Basis\Nats\Queue;
+
+class ServiceEndpoint
+{
+    private int $num_requests = 1;
+
+    private int $num_errors = 0;
+
+    private string $last_error = '';
+
+    private float $processing_time = 0;
+
+    private Client|Queue $subscription;
+
+    public function __construct(
+        private readonly Service $service,
+        private readonly string  $name,
+        private readonly string  $subject,
+        private                  $endpointHandler,
+        private readonly string  $queue_group = 'q'
+    )
+    {
+        $this->subscription = $this->service->client->subscribeQueue(
+            $this->subject,
+            $this->queue_group,
+            function (Payload $message) {
+                // Start calculating the time
+                $start = microtime(true);
+
+                // Update the endpoint metrics
+                $this->num_requests += 1;
+
+                // Setup the response
+                $response = "";
+
+                switch ($this->endpointHandler) {
+                    case is_string($this->endpointHandler):
```

Comment:
let's replace this case with `is_subclass_of` check and add default case with logic exception